components/director/internal/domain/label/repository_test.go

```diff
@@ -511,7 +511,202 @@ func TestRepository_List(t *testing.T) {
+		mockConverter.On("FromEntity", inputItems[0]).Return(*expected[0], nil).Once()
+		mockConverter.On("FromEntity", inputItems[1]).Return(*expected[1], nil).Once()
+
+		repo := label.NewRepository(mockConverter)
+
+		db, dbMock := mockDatabase(t)
+		defer func() {
+			require.NoError(t, dbMock.ExpectationsWereMet())
+		}()
+
+		escapedQuery := regexp.QuoteMeta(`SELECT "id", "tenant_id", "key", "value", "app_id", "runtime_id" FROM "public"."labels" WHERE "key" = $1 AND "tenant_id" = $2`)
+		mockedRows := sqlmock.NewRows([]string{"id", "tenant_id", "key", "value", "app_id", "runtime_id"}).
+			AddRow("1", tnt, labelKey, "test1", appObjID, nil).
+			AddRow("2", tnt, labelKey, "test2", secondAppObjID, nil)
+		dbMock.ExpectQuery(escapedQuery).WithArgs(labelKey, tnt).WillReturnRows(mockedRows)
+
+		ctx := context.TODO()
+		ctx = persistence.SaveToContext(ctx, db)
+		// WHEN
+		actual, err := repo.ListByKey(ctx, tnt, labelKey)
+		// THEN
+		require.NoError(t, err)
+		assert.ElementsMatch(t, expected, actual)
+	})
+
+	t.Run("Success - Labels only for Runtimes", func(t *testing.T) {
+		// GIVEN
+		tnt := "tenant"
+		labelKey := "foo"
+		rtmObjType := model.RuntimeLabelableObject
+
+		rtmObjID := "foo"
+		secondRtmObjID := "bar"
+
+		inputItems := []label.Entity{
+			{ID: "1", TenantID: tnt, Key: labelKey, Value: "test1", AppID: sql.NullString{}, RuntimeID: sql.NullString{Valid: true, String: rtmObjID}},
+			{ID: "2", TenantID: tnt, Key: labelKey, Value: "test2", AppID: sql.NullString{}, RuntimeID: sql.NullString{Valid: true, String: secondRtmObjID}},
+		}
+		expected := []*model.Label{
+			{ID: "1", Tenant: tnt, Key: labelKey, Value: "test1", ObjectType: rtmObjType, ObjectID: rtmObjID},
+			{ID: "2", Tenant: tnt, Key: labelKey, Value: "test2", ObjectType: rtmObjType, ObjectID: secondRtmObjID},
+		}
+
+		mockConverter := &automock.Converter{}
+		defer mockConverter.AssertExpectations(t)
+		mockConverter.On("FromEntity", inputItems[0]).Return(*expected[0], nil).Once()
+		mockConverter.On("FromEntity", inputItems[1]).Return(*expected[1], nil).Once()
+
+		repo := label.NewRepository(mockConverter)
+
+		db, dbMock := mockDatabase(t)
+		defer func() {
+			require.NoError(t, dbMock.ExpectationsWereMet())
+		}()
+
+		escapedQuery := regexp.QuoteMeta(`SELECT "id", "tenant_id", "key", "value", "app_id", "runtime_id" FROM "public"."labels" WHERE "key" = $1 AND "tenant_id" = $2`)
+		mockedRows := sqlmock.NewRows([]string{"id", "tenant_id", "key", "value", "app_id", "runtime_id"}).
+			AddRow("1", tnt, labelKey, "test1", nil, rtmObjID).
+			AddRow("2", tnt, labelKey, "test2", nil, secondRtmObjID)
+		dbMock.ExpectQuery(escapedQuery).WithArgs(labelKey, tnt).WillReturnRows(mockedRows)
+
+		ctx := context.TODO()
+		ctx = persistence.SaveToContext(ctx, db)
+		// WHEN
+		actual, err := repo.ListByKey(ctx, tnt, labelKey)
+		// THEN
+		require.NoError(t, err)
+		assert.ElementsMatch(t, expected, actual)
+	})
+
+	t.Run("Error - Doesn't exist", func(t *testing.T) {
```

Comment:
`Error - Doesn't exist`
and 
```
require.NoError(t, err)
assert.Empty(t, actual)
```

make it consitent

Reply:
done